sys/log/full/src/log_fcb.c

```diff
@@ -102,17 +132,24 @@ log_fcb_find_gte(struct log *log, struct log_offset *log_offset,
     }
 #endif
 
-    /* Keep advancing until we find an entry with a great enough index. */
-    do {
-        rc = log_read_hdr(log, out_entry, &hdr);
-        if (rc != 0) {
-            return rc;
-        }
+    if (fcb->f_oldest == fcb->f_active.fe_area) {
+        fap = fcb->f_oldest;
+    } else {
+        /* start from one before */
+        fap = (fcb->f_active.fe_area - sizeof(struct flash_area));
+    }
 
-        if (hdr.ue_index >= log_offset->lo_index) {
-            return 0;
-        }
-    } while (fcb_getnext(fcb, out_entry) == 0);
+    lwga.log            = log;
+    lwga.log_offset     = log_offset;
+    lwga.fcb_entry      = out_entry;
+    rc = fcb_walk(fcb, fap, fcb_walk_gte, (void *)&lwga);
+    /* Not found, try the active area */
+    if (rc == 0) {
+        fap = fcb->f_active.fe_area;
+        rc  = fcb_walk(fcb, fap, fcb_walk_gte, (void *)&lwga);
+    }
+
+    assert(rc == FCB_FOUND);
```

Comment:
expect to find every time, not sure assert is the right thing to do here though.

Reply:
or return an error